charms/reactive/flags.py

```diff
@@ -281,6 +308,7 @@ def get_state(flag, default=None):
 
 @hookenv.atstart
 def _manage_automatic_flags():
+    _clear_triggers()
```

Comment:
Yeah, that is a bug here.  However, I added this because I realized that the triggers are actually persisted between hook invocations since they're stored in unitdata and they get duplicates added, ballooning the data, and potentially interfering with unregistering a trigger.  I think this could be resolved by moving it to `atexit` instead, though?